docs/release_cycle.md

```diff
@@ -0,0 +1,72 @@
+# Release Cycle
+
+## Schedule
+
+These are the upcoming releases dates:
+
+| Release | Release Planning Meeting | Release Candidate 1   | General Availability    | Release Champion | Checklist                                                 |
+|---------|--------------------------|-----------------------|-------------------------|------------------|-----------------------------------------------------------|
+| v2.6    | Monday, Nov. 28, 2022    | Monday, Dec. 12, 2022 | Tuesday, Jan. 2, 2023   |                  | [checklist](https:/github.com/argoproj/releases/issues/1) |
+| v2.7    | Monday, Feb. 27, 2023    | Monday, Mar. 13, 2023 | Monday, Apr. 3, 2023    |
+| v2.8    | Monday, May 29, 2023     | Monday, Jun. 12, 2023 | Wednesday, Jul. 5, 2023 |
+| v2.9    | Monday, Aug. 28, 2023    | Monday, Sep. 11, 2023 | Monday, Oct. 2, 2023    |
+
+## Release Process
+
+### Minor Releases (e.g. 2.x.0)
+
+A minor Argo CD release occurs four times a year, once every three months. Each General Availability (GA) release is 
+preceded by several Release Candidates (RCs). The first RC is released three weeks before the scheduled GA date. This 
+effectively means that there is a three-week feature freeze.
+
+These are the approximate release dates:
+
+* The first Monday of January
+* The first Monday of April
+* The first Monday of July
+* The first Monday of October
+
+Dates may be shifted slightly to accommodate holidays. Those shifts should be minimal.
+
+### Patch Releases (e.g. 2.5.x)
+
+Argo CD patch releases occur on an as-needed basis. Only the three most recent minor versions are eligible for patch
+releases. Versions older than the three most recent minor versions are considered EOL and will not receive bug fixes or
+security updates.
+
+### Minor Release Planning Meeting
+
+Two weeks before the RC date, there will be a meeting to discuss which features are planned for the RC. This meeting is
+for contributors to advocate for certain features. Features which have at least one approver (besides the contributor) 
+who can assure they will review/merge by the RC date will be included in the release milestone. All other features will
+be dropped from the milestone (and potentially shifted to the next one).
+
+Since not everyone will be able to attend the meeting, there will be a meeting doc. Contributors can add their feature
+to a table, and approvers can add their name to the table. Features with a corresponding approver will remain in the 
+release milestone.
+
+### Release Champion
```

Comment:
Who all can be release champion? Can reviewers be release champion? As a reviewer I don't have write access to the repo

Reply:
I think they would have to be at least approver in order to cut the RC.

But ideally I think this role could be literally anyone who's willing to manage the checklist. They could be in charge of delegating the tasks to folks with the necessary permissions. I'll update the checklist/ PR to reflect that.

Reply:
Added a note.